Simplified/LibraryFinder/UI/NYPLLibraryFinderLibraryCell.swift

```diff
@@ -0,0 +1,203 @@
+//  Simplified
+//
+//  Created by Ernest Fan on 2021-04-22.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import UIKit
+
+enum NYPLLibraryFinderLibraryCellType {
+  case myLibrary
+  case newLibrary
+}
+
+class NYPLLibraryFinderLibraryCell: UICollectionViewCell {
+  static let reuseIdentifier = "NYPLLibraryFinderLibraryCellReuseIdentifier"
+  
+  private let stackViewSpacing: CGFloat = 4.0
+  
+  private var badgeLabelWidthConstraint: NSLayoutConstraint?
+  private var badgeLabelHeightConstraint: NSLayoutConstraint?
+  
+  private var type: NYPLLibraryFinderLibraryCellType = .myLibrary
+  
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    
+    setupUI()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  override func layoutSubviews() {
+    super.layoutSubviews()
+    // Redraw the border when the height of the cell changes
+    updateBorder()
+  }
+  
+  override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+    super.traitCollectionDidChange(previousTraitCollection)
+    // Redraw the border when dark mode activate/deactivate
+    updateBorder()
+  }
+  
+  override func systemLayoutSizeFitting(_ targetSize: CGSize, withHorizontalFittingPriority horizontalFittingPriority: UILayoutPriority, verticalFittingPriority: UILayoutPriority) -> CGSize {
+    // The vertical fitting priority is
+    // .fittingSizeLevel meaning the cell will find the
+    // height that best fits the content
+    let size = super.systemLayoutSizeFitting(
+      targetSize,
+      withHorizontalFittingPriority: .required,
+      verticalFittingPriority: .fittingSizeLevel
+    )
+    
+    return size
+  }
+  
+  // MARK: - UI
+  
+  func configureCell(type: NYPLLibraryFinderLibraryCellType, account: Account) {
+    self.type = type
+    configureBorder(for: type)
+    
+    nameLabel.text = account.name
+    // TODO: iOS-36 Assign data value to UI componenets after parser and data model are updated
+    descriptionLabel.text = account.subtitle
+    distanceLabel.text = "20 Mi"
+    postalCodeLabel.text = "12345"
+    
+    configureBadgeLabel(text: "BADGE")
+  }
+  
+  private func setupUI() {
+    backgroundColor = NYPLLibraryFinderConfiguration.cellBackgroundColor
+    layer.cornerRadius = NYPLLibraryFinderConfiguration.cellCornerRadius(type: self.type)
+    
+    let leftStackView = UIStackView(arrangedSubviews: [nameLabel, descriptionLabel])
+    leftStackView.axis = .vertical
+    leftStackView.spacing = stackViewSpacing
+    leftStackView.alignment = .leading
+    
+    let rightStackView = UIStackView(arrangedSubviews: [distanceLabel, postalCodeLabel, UIView()])
+    rightStackView.axis = .vertical
+    rightStackView.spacing = stackViewSpacing
+    rightStackView.alignment = .trailing
+    
+    addSubview(divider)
+    addSubview(leftStackView)
+    addSubview(rightStackView)
+    addSubview(badgeLabel)
+    
```

Comment:
should we add these views during lazy initialization so that we don't accidentally add the same view twice?